crates/rome_js_analyze/src/semantic_analyzers/style/no_static_only_class.rs

```diff
@@ -0,0 +1,164 @@
+                    AnyJsClassMember::JsPropertyClassMember(m) => m
+                        .modifiers()
+                        .iter()
+                        .any(|modifier| modifier.as_js_static_modifier().is_some()),
+                    AnyJsClassMember::JsSetterClassMember(m) => m
+                        .modifiers()
+                        .iter()
+                        .any(|modifier| modifier.as_js_static_modifier().is_some()),
+                    AnyJsClassMember::JsStaticInitializationBlockClassMember(_) => true, // See GH#4482: Static initialization blocks are regarded as static
+                    AnyJsClassMember::TsConstructorSignatureClassMember(_) => false, // TODO: What to do with TS types?
+                    AnyJsClassMember::TsGetterSignatureClassMember(m) => m
+                        .modifiers()
+                        .iter()
+                        .any(|modifier| modifier.as_js_static_modifier().is_some()),
+                    AnyJsClassMember::TsIndexSignatureClassMember(m) => m
+                        .modifiers()
+                        .iter()
+                        .any(|modifier| modifier.as_js_static_modifier().is_some()),
+                    AnyJsClassMember::TsInitializedPropertySignatureClassMember(m) => m
+                        .modifiers()
+                        .iter()
+                        .any(|modifier| modifier.as_js_static_modifier().is_some()),
+                    AnyJsClassMember::TsMethodSignatureClassMember(m) => m
+                        .modifiers()
+                        .iter()
+                        .any(|modifier| modifier.as_js_static_modifier().is_some()),
+                    AnyJsClassMember::TsPropertySignatureClassMember(m) => m
+                        .modifiers()
+                        .iter()
+                        .any(|modifier| modifier.as_js_static_modifier().is_some()),
+                    AnyJsClassMember::TsSetterSignatureClassMember(m) => m
+                        .modifiers()
+                        .iter()
+                        .any(|modifier| modifier.as_js_static_modifier().is_some()),
+                });
+        if all_members_static {
+            Some(())
+        } else {
+            None
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let class_declaration = ctx.query();
+        let class_keyword = class_declaration.class_token().ok()?;
+
+        let diagnostic_node = class_declaration
+            .id()
+            .ok()?
+            .map_or(class_keyword.text_trimmed_range(), |id| {
+                id.syntax().text_trimmed_range()
+            });
+
+        // TODO: Wording
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                diagnostic_node,
+                markup! {
+                    "Don't use static classes as namespaces."
```

Comment:
Maybe something like:

```suggestion
                    "A class that includes only "<Emphasis>"static members"</Emphasis>" is confusing."
```

What do you think @ematipico?

Reply:
Considering that this rule is more about style, I'd suggest a different wording:

"Avoid classes that contain only static fields"

And then we can add a note saying "Prefer using simple functions instead of classes with only static fields"

Reply:
Looks fair. I would prefer "members" instead of "fields" in the emssage.